File: src/main/java/com/example/solidaryaction/entities/Doador.java
package com.example.solidaryaction.entities;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Doador {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String nome;

    public String cpf;

    public String email;

    public String profissao;

    public String endereco;
}

File: front/app/types/doador.ts
export class Doador{
    constructor(
        public id: number | null,
        public nome:string,
        public email:string,
        public profissao:string,
        public endereco:string,
        public status:string,
        public cpf:string
    ){}
}

export interface DoadorFormProps{
    doadorExistente?:Doador
}